homeassistant/components/roborock/coordinator.py

```diff
@@ -288,11 +289,16 @@ async def _async_update_data(self) -> DeviceProp:
 
             # If the vacuum is currently cleaning and it has been IMAGE_CACHE_INTERVAL
             # since the last map update, you can update the map.
-            if (
-                self.current_map is not None
-                and self.roborock_device_info.props.status.in_cleaning
-                and (dt_util.utcnow() - self.maps[self.current_map].last_updated)
-                > IMAGE_CACHE_INTERVAL
+            if self.current_map is not None and (
+                (
+                    self.roborock_device_info.props.status.in_cleaning
+                    and (dt_util.utcnow() - self.maps[self.current_map].last_updated)
+                    > IMAGE_CACHE_INTERVAL
+                )
+                or (
+                    previous_state != self.roborock_device_info.props.status.state_name
+                    and previous_state is not None
+                )
```

Comment:
Can `previous_state is not None` be omitted? That seems to indicate it would be the first refresh, which seems fine to do a map update. Also extracting a local status variable can help avoid some wrapping that might benefit readability:

```suggestion
            new_state = self.roborock_device_info.props.status
            if self.current_map is not None and (
                (
                    new_state.in_cleaning
                    and (dt_util.utcnow() - self.maps[self.current_map].last_updated)
                    > IMAGE_CACHE_INTERVAL
                )
                or previous_state != new_state.state_name
```

Reply:
This does make a small change to a test - but i think it's fine as it is a logical improvement.